scripts/create-darwin-volume.sh

```diff
@@ -435,12 +435,21 @@ EOF
     return 1
 }
 
-add_nix_vol_fstab_line() {
+# intentional subshell function to make exporting these envs w/o leaking easier
+add_nix_vol_fstab_line() (
     local uuid="$1"
     # shellcheck disable=SC1003,SC2026
     local escaped_mountpoint="${NIX_ROOT/ /'\\\'040}"
     shift
-    EDITOR="/usr/bin/ex" _sudo "to add nix to fstab" "$@" <<EOF
+    # exporting vim variables to work around a problem with vim plugins
+    # breaking the exit codes; see https://github.com/NixOS/nix/issues/5468
+    # we'd prefer --noplugin flag as elsewhere, but vifs doesn't word-split
+    # the EDITOR env.
+    # VIM == keep it from loading system plugins
+    # HOME == keep it from loading user plugins
+    # EDITOR == just telling vifs which editor to use
+    export VIM=/dev/null HOME= EDITOR="/usr/bin/ex"
```

Comment:
This is a little dumb, so I'm happy if someone else who knows vim better can find a cleaner solution. 

We're exporting editor so that `vifs` will use it, but including flags in the env didn't seem to work (it doesn't appear to word-split the env). Vim makes it easy to disable this with either the -u NONE or --noplugin flags, but we can't use them here. I tried something like `export VIMINIT="set noloadplugins"`, but I couldn't get that working (I saw some mention that makes me think this may happen _too early_ to matter here?)

Setting VIM and HOME is just making it search the wrong locations for the plugins.

You can play with this behavior like:
```console
$ echo blah | ex -V10
Searching for "pack/*/start/*" in "/Users/abathur/.vim,/dev/null/vimfiles,/dev/null,/dev/null/vimfiles/after,/Users/abathur/.vim/after"
Searching for "/Users/abathur/.vim/pack/*/start/*"
Searching for "/dev/null/vimfiles/pack/*/start/*"
Searching for "/dev/null/pack/*/start/*"
Searching for "/dev/null/vimfiles/after/pack/*/start/*"
Searching for "/Users/abathur/.vim/after/pack/*/start/*"
not found in 'packpath': "pack/*/start/*"
Searching for "plugin/**/*.vim" in "/Users/abathur/.vim,/dev/null/vimfiles,/dev/null,/dev/null/vimfiles/after,/Users/abathur/.vim/after"
Searching for "/Users/abathur/.vim/plugin/**/*.vim"
Searching for "/dev/null/vimfiles/plugin/**/*.vim"
Searching for "/dev/null/plugin/**/*.vim"
not found in 'packpath': "plugin/**/*.vim"
Searching for "pack/*/start/*" in "/Users/abathur/.vim,/dev/null/vimfiles,/dev/null,/dev/null/vimfiles/after,/Users/abathur/.vim/after"
Searching for "/Users/abathur/.vim/pack/*/start/*"
Searching for "/dev/null/vimfiles/pack/*/start/*"
Searching for "/dev/null/pack/*/start/*"
Searching for "/dev/null/vimfiles/after/pack/*/start/*"
Searching for "/Users/abathur/.vim/after/pack/*/start/*"
not found in 'packpath': "pack/*/start/*"
Searching for "plugin/**/*.vim" in "/Users/abathur/.vim,/dev/null/vimfiles,/dev/null,/dev/null/vimfiles/after,/Users/abathur/.vim/after"
Searching for "/dev/null/vimfiles/after/plugin/**/*.vim"
Searching for "/Users/abathur/.vim/after/plugin/**/*.vim"
not found in 'runtimepath': "plugin/**/*.vim" 
Entering Ex mode.  Type "visual" to go to Normal mode.
:blah
E492: Not an editor command: blah

$ echo blah | ex --noplugin -V10
Entering Ex mode.  Type "visual" to go to Normal mode.
:blah
E492: Not an editor command: blah
```

Reply:
Replaced this process with a shell wrapper.